<?xml version="1.0"?>
<robot name="depthai_camera" xmlns:xacro="http://ros.org/wiki/xacro">

    <xacro:arg name="camera_name"   default="oak" />
    <xacro:arg name="camera_model"  default="OAK-D" />
    <xacro:arg name="base_frame"    default="oak-d_frame" />
    <xacro:arg name="parent_frame"  default="oak-d-base-frame" />
    <xacro:arg name="cam_pos_x"     default="0.0" />
    <xacro:arg name="cam_pos_y"     default="0.0" />
    <xacro:arg name="cam_pos_z"     default="0.0" />
    <xacro:arg name="cam_roll"      default="0.0" />
    <xacro:arg name="cam_pitch"     default="0.0" />
    <xacro:arg name="cam_yaw"       default="0.0" />

    <xacro:include filename="$(find depthai_bridge)/urdf/include/depthai_macro.urdf.xacro"/>

    <link name="$(arg parent_frame)"/>
    <xacro:depthai_camera camera_name = "$(arg camera_name)" parent = "$(arg parent_frame)" camera_model = "$(arg camera_model)" base_frame = "$(arg base_frame)" cam_pos_x = "$(arg cam_pos_x)" cam_pos_y = "$(arg cam_pos_y)" cam_pos_z = "$(arg cam_pos_z)" cam_roll = "$(arg cam_roll)" cam_pitch = "$(arg cam_pitch)" cam_yaw = "$(arg cam_yaw)"/>

</robot>
